### ISSUE TITLE
Create activity log for an agent

### ISSUE THREAD

**Issue by gabrielfior** [gnosis/prediction-market-agent#61]
See previous PR #48 for a previous discussion on this topic. Main idea is this:
```
- get activity log (buy, sell) ?? -> (GF) It would be interesting to have the agent
keep an internal orderbook of sorts, and in the very beginning (and in the end)
of `run` the agent prints out the orders it made.
```

Specifically for this feature, an orderbook can be constructed by querying the Omen-xDAI subgraph and fetching the trades it executed (via `OmenSubgraphHandler.get_bets`). It would be nice to arrange those bets in a way that the outcome becomes clear, so that we can query the amount of tokens the agent owns for each outcome.

Other options include having a DB for this - also possible, but I think it's a bit of an overkill for this specific feature.

Calling @kongzii and @evangriffiths for feedback and task definition.

**Comment by kongzii** [comment]
If I understand correctly, do you want to print out something like what we have in the monitoring app? If so, what is the motivation for printing it? If not, what would be different?

![Screenshot by Dropbox Capture](https://github.com/gnosis/prediction-market-agent/assets/15619339/d574e367-b7c4-4bf3-a89b-474552f9fa87)

And Dune dashboard has something similar as well:

![Screenshot by Dropbox Capture](https://github.com/gnosis/prediction-market-agent/assets/15619339/d4c1e8bc-3f1c-4045-8dad-4e57d4c0adbe)

**Comment by kongzii** [comment]
> and in the very beginning (and in the end) of `run` the agent prints out the orders it made.

I thought you just wanted to print it out to logs, but I can see having something like `my orders` function, and then he can decide if there is something to sell, buy more, use for a different prediction, etc. etc. 👍

**Comment by evangriffiths** [comment]
> I thought you just wanted to print it out to logs, but I can see having something like my orders function, and then he can decide if there is something to sell, buy more, change opinion, use for a different prediction, etc. etc. 👍

Yeah that's the motivation for this issue. I guess it needs experimenting with to see how it gets used and if it's useful. e.g. the agent might call it before doing `SummarizeLearning`.

> I thought you just wanted to print it out to logs
I think there's also this... a general agent is different to a `DeployableAgent`, because it's not as constrained. So the monitoring table/dune table doesn't really account for all its activity. I think you also want to see the reasoning steps too, i.e.
```
How much is (2*4 + 3)*5?

Running 10 iterations
>> Reasoning("I need to reason step-by-step")
The reasoning has been recorded
>> Reasoning("First, calculate the product of 2 and 4")
The reasoning has been recorded
>> Product(a=2, b=4)
8
>> Reasoning("Then, add 3 to the product of 2 and 4")
The reasoning has been recorded
>> Sum(a=8, b=3)
11
>> Reasoning("Lastly, multiply the sum by 5")
The reasoning has been recorded
>> Product(a=11, b=5)
55
>> Reasoning("So, the result of (2*4 + 3)*5 is 55")
The reasoning has been recorded
>> Stop()
The program has been stopped
```

A log of every function called and its return value is already conveniently stored in `agent.history`. So maybe the deployed agent can store this, and the agent's PMA dir we can write a streamlit script for displaying this nicely for a deployed agent.

**Comment by gabrielfior** [comment]
This can be closed by #122 , since this information is now retrievable by the agent.
